src/code_28910_len_5090.c

```diff
@@ -84,63 +196,248 @@ INCLUDE_ASM(s32, "code_28910_len_5090", func_8004EC04);
 
 INCLUDE_ASM(s32, "code_28910_len_5090", func_8004EC68);
 
-INCLUDE_ASM(s32, "code_28910_len_5090", func_8004FBBC);
+void func_8004FBBC(UnkAlA9C* arg0, UnkAl60* arg1) {
+    u32 unk_D4 = arg0->unk_D4.u16;
+    s32 temp_v0;
+
+    arg0->unk_208 = unk_D4;
+    temp_v0 = func_8004FC08(arg0, unk_D4);
+    arg0->unk_B0 = temp_v0;
+    arg0->unk_08 = temp_v0 * 10;
+    arg0->unk_BC = 0;
+    arg0->unk_B8 = 0;
+    arg0->unk_B4 = 0;
+}
+
+s32 func_8004FC08(UnkAlA9C* arg0, u32 arg1) {
+    u32 t32;
+    u16 t16;
+    u32 ret;
+
+    t32 = t16 = arg0->unk_20A;
+
+    ret = arg1;
+    ret *= arg0->unk_D0;
```

Comment:
can this potentially be rewritten as ``ret = arg1 * arg0->unk_D0;`` ?

Reply:
Unfortunately no, that screws with reg alloc.

Reply:
rip